MovieMVP/MovieMVPTests/MockNetworkSevice/MockNetworkService.swift

```diff
@@ -0,0 +1,73 @@
+// MockNetworkService.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+@testable import MovieMVP
+import SwiftyJSON
+
+/// Мок сетевого слоя
+final class MockNetworkService: NetworkServiceProtocol {
+    // MARK: - Public Properties
+
+    var movies: [Movie]!
+    var movieDetail: MovieDetails!
+
+    // MARK: - Init
+
+    init() {}
+
+    convenience init(movies: [Movie]?) {
+        self.init()
+        self.movies = movies
+    }
+
+    convenience init(movieDetail: MovieDetails?) {
+        self.init()
+        self.movieDetail = movieDetail
+    }
+
+    // MARK: - Public Methods
+
+    func fetchMovies(category: MovieMVP.Category, completion: @escaping (Result<[Movie], Error>) -> Void) {
+        guard var movies = movies else {
+            let error = NSError(domain: "", code: 0)
+            completion(.failure(error))
+            return
+        }
+        let json = getData(name: "MockMovie")
+        let movieJSON = json["results"].arrayValue.map { Movie(json: $0) }
+        movies = movieJSON
+        completion(.success(movies))
+    }
+
+    func fetchMovieDetails(id: Int, completion: @escaping (Result<MovieDetails, Error>) -> Void) {
+        if var movieDetail = movieDetail {
+            let json = getData(name: "MockMovieDetails")
+            let movieDetailJSON = MovieDetails(json: json)
+            movieDetail = movieDetailJSON
+            completion(.success(movieDetail))
+        } else {
+            let error = NSError(domain: "", code: 0)
+            completion(.failure(error))
+        }
+    }
+
+    // MARK: - Public Methods
+
+    private func getData(name: String, withExtension: String = "json") -> JSON {
+        guard
+            let jsonURL = Bundle.main.path(forResource: name, ofType: withExtension)
+        else {
+            return JSON()
+        }
+        do {
+            let fileURL = URL(fileURLWithPath: jsonURL)
+            let data = try Data(contentsOf: fileURL)
+            let json = try JSON(data: data)
+            return json
+        } catch {
+            print("error")
```

Comment:
```suggestion
            print("error")
```
в константы или error.localizedDescription